### PR TITLE
[SD-3788] - Relating a story as an update (rewrite) of another

### PR DESCRIPTION
**Pull request by akintolga**

### CODE REVIEW

**Comment by petrjasek** [comment]
:+1:

File: apps/archive/archive_rewrite.py [hunk 1]
```diff
@@ -78,28 +91,53 @@ def _validate_rewrite(self, original):
         if not TakesPackageService().is_last_takes_package_item(original):
             raise SuperdeskApiError.badRequestError(message="Only last take of the package can be rewritten.")
 
-    def _create_rewrite_article(self, original, digital):
+        if update:
+            if update.get('rewrite_of'):
+                raise SuperdeskApiError.badRequestError("Rewrite story has been used as update before !")
+
+            if update.get(ITEM_STATE) in ['published', 'corrected', 'killed', 'scheduled', 'spiked']:
```
Comment (sivakuna-aap) [inline apps/archive/archive_rewrite.py]: Use `CONTENT_STATE` constant

File: apps/archive/archive_rewrite.py [hunk 2]
```diff
@@ -78,28 +91,53 @@ def _validate_rewrite(self, original):
...
+                raise InvalidStateTransitionError()
+
+            if update.get(ITEM_TYPE) not in [CONTENT_TYPE.TEXT, CONTENT_TYPE.PREFORMATTED]:
+                raise SuperdeskApiError.badRequestError("Rewrite story can only be text or preformatted !")
+
+            if update.get(ITEM_TYPE) == CONTENT_TYPE.COMPOSITE:
```
Comment (sivakuna-aap) [inline apps/archive/archive_rewrite.py]: I see this as unreachable code. If a request comes with type as composite the if condition in line 101 will catch it.

Comment (akintolga) [inline apps/archive/archive_rewrite.py]: Good catch Siva, thanks.

**Comment by mdhaman** [comment]
:+1:

**Comment by ioanpocol** [comment]
:+1:

**Pull request merged**